widgets/CloudStorageToSnowflake-action.json

```diff
@@ -0,0 +1,550 @@
+          "name": "sourceQuery"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Destination Table",
+          "name": "destinationTable"
+        }
+      ]
+    },
+    {
+      "label": "Credentials",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Username",
+          "name": "username"
+        },
+        {
+          "widget-type": "password",
+          "label": "Password",
+          "name": "password"
+        }
+      ]
+    },
+    {
+      "label": "Key Pair Authentication",
+      "properties": [
+        {
+          "widget-type": "toggle",
+          "label": "Key Pair Authentication Enabled",
+          "name": "keyPairEnabled",
+          "widget-attributes": {
+            "default": "false",
+            "on": {
+              "value": "true",
+              "label": "True"
+            },
+            "off": {
+              "value": "false",
+              "label": "False"
+            }
+          }
+        },
+        {
+          "widget-type": "textarea",
+          "label": "Private Key",
+          "name": "privateKey"
+        },
+        {
+          "widget-type": "password",
+          "label": "Key File Passphrase",
+          "name": "passphrase"
+        }
+      ]
+    },
+    {
+      "label": "OAuth2",
+      "properties": [
+        {
+          "widget-type": "toggle",
+          "label": "OAuth2 Enabled",
+          "name": "oauth2Enabled",
+          "widget-attributes": {
+            "default": "false",
+            "on": {
+              "value": "true",
+              "label": "True"
+            },
+            "off": {
+              "value": "false",
+              "label": "False"
+            }
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Client ID",
+          "name": "clientId"
+        },
+        {
+          "widget-type": "password",
+          "label": "Client Secret",
+          "name": "clientSecret"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Refresh Token",
+          "name": "refreshToken"
+        }
+      ]
+    },
+    {
+      "label": "Cloud Provider Parameters",
+      "properties": [
+        {
+          "widget-type": "toggle",
+          "label": "Use Cloud Provider Parameters",
+          "name": "useCloudProviderParameters",
+          "widget-attributes": {
+            "default": "false",
+            "on": {
+              "value": "true",
+              "label": "True"
+            },
+            "off": {
+              "value": "false",
+              "label": "False"
+            }
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Cloud Provider",
+          "name": "cloudProvider",
+          "widget-attributes": {
+            "values": [
+              "NONE",
+              "GCP",
+              "AWS",
+              "Microsoft Azure"
```

Comment:
These  values are different than written in the plugin doc - `**Cloud Provider:** Cloud provider name. Possible values: `azure`, `gcs`, `s3`.`